docs/new_services.rst

```diff
@@ -19,3 +23,131 @@ Each service is implemented as a class and should at least implement the followi
+
+.. _service-example-ref:
+
+New Service Example
+===================
+
+Let’s walk through an example of writing a simple Zookeeper service.
+
+.. code-block:: python
+
+    class ZookeeperService(Service):
+        PERSISTENT_ROOT = "/mnt"
+        LOG_FILE = os.path.join(PERSISTENT_ROOT, "zk.log")
+        DATA_DIR = os.path.join(PERSISTENT_ROOT, "zookeeper")
+        CONFIG_FILE = os.path.join(PERSISTENT_ROOT, "zookeeper.properties")
+
+        logs = {
+            "zk_log": {
+                "path": LOG_FILE,
+                "collect_default": True},
+            "zk_data": {
+                "path": DATA_DIR,
+                "collect_default": False}
+        }
+
+        def __init__(self, context, num_nodes):
+            super(ZookeeperService, self).__init__(context, num_nodes)
+
+
+Log files will be collected on both successful and failed test runs, while files from the data directory will be collected only on failed test runs. Zookeeper service requests the number of nodes passed to its constructor by passing ``num_nodes`` parameters to the Service base class’s constructor.
+
+.. code-block:: python
+
+        def start_node(self, node):
+            idx = self.idx(node)
+            self.logger.info("Starting ZK node %d on %s", idx, node.account.hostname)
+
+            node.account.ssh("mkdir -p %s" % self.DATA_DIR)
+            node.account.ssh("echo %d > %s/myid" % (idx, self.DATA_DIR))
+
+            prop_file = """\n dataDir=%s\n clientPort=2181""" % self.DATA_DIR
+            for idx, node in enumerate(self.nodes):
+                prop_file += "\n server.%d=%s:2888:3888" % (idx, node.account.hostname)
+            self.logger.info("zookeeper.properties: %s" % prop_file)
+            node.account.create_file(self.CONFIG_FILE, prop_file)
+
+            start_cmd = "/opt/kafka/bin/zookeeper-server-start.sh %s 1>> %s 2>> %s &" % \
+                    (self.CONFIG_FILE, self.LOG_FILE, self.LOG_FILE)
+
+            with node.account.monitor_log(self.LOG_FILE) as monitor:
+                node.account.ssh(start_cmd)
+                monitor.wait_until(
+                    "binding to port",
+                    timeout_sec=100,
+                    backoff_sec=7,
+                    err_msg="Zookeeper service didn't finish startup"
+                )
+            self.logger.debug("Zookeeper service is successfully started.")
+
+
+The ``start_node`` method first creates directories and the config file on the given node, and then invokes the start script to start a Zookeeper service. In this simple example, the config file is created from ``prop_file`` string. You can also create config file from a template, as described in :ref:`using-templates-ref`.
```

Comment:
from ``prop_file`` string -> from a manually constructed ``prop_file`` string because it has only a couple of easy to construct lines. More complex config files can be created with templates...